ArtOfIllusion/src/artofillusion/LayoutWindow.java

```diff
@@ -1344,6 +1346,13 @@ public void setObjectListVisible(boolean visible)
     setDockableWidgetVisible((DockableWidget) itemTreeScroller.getParent(), visible);
   }
 
+  /** Set whether the user has set the object to be displayed. */
+
+  public boolean isObjectListVisible()
```

Comment:
Why do we need a public getter here? This state is only relevant for the internals of Layout window.

I'm actually unsure why `setObjectListVisible()` is public either... I don't think that it is used outside of the class.

Reply:
> Why do we need a public getter here? / ... / I'm actually unsure why `setObjectListVisible()` is public either... I don't think that it is used outside of the class. ... 

For a moment I wondered that myself, but all the other similar things are public as well. Now this _almost_ enables external control, but a little something was needed to have that happen properly. As I said: "added for a keyboard shortcut." Went to preferences and added a toggle to the letter `L` (list) -- My other candidate  would have been `I` (item list). 

Looks like this in the xml file.

~~~~xml
<keystroke language="Groovy" modifiers="0" code="76" name="Hide/Show Object List">if (window instanceof LayoutWindow)((LayoutWindow)window).setObjectListVisible(!((LayoutWindow)window).isObjectListVisible());</keystroke>
~~~~

To a _"heavy-user"_ _(a favorite term of my previous employers)_ that would be handy and actually I'd like to add this arrangement for the score too.

The obvious problem of course is, that users probably will not know about the shortcut keys, that are not suggested in the menu item prompts and an unfortunately there is no way of adding single key shortcuts (without modifiers) thru translations so they'd become visible. --> Introduction video needed.

Reply:
>unfortunately there is no way of adding single key shortcuts (without modifiers) thru translations so they'd become visible.

There is, however, an overloaded version of the `Translate.menuItem()` method that takes a `buoy.widget.Shortcut`. I think that it should be possible to hard-code a shortcut with no modifiers. We might even be able to localize with a little more code added. This would be my preferred solution.

Reply:
To be clear:

Adding a shortcut key/pattern for this operation is a separate matter. I'd rather not add the getter, because then we kind of have to support it, even if we do a better fix later.

Reply:
pushing

Reply:
ok